include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -975,6 +975,29 @@ def RegToMem : Pass<"regtomem", "mlir::func::FuncOp"> {
   let dependentDialects = ["cudaq::cc::CCDialect", "quake::QuakeDialect"];
 }
 
+def RemoveUselessStores : Pass<"remove-useless-stores", "mlir::func::FuncOp"> {
```

Comment:
I think we should call this "write-after-write-elimination`.

We can also make it a generic pass, since we may want to run it on more than `func.func`.